@xen-orchestra/backups/_cleanVm.js

```diff
@@ -31,66 +31,44 @@ const computeVhdsSize = (handler, vhdPaths) =>
     }
   )
 
-// chain is an array of VHDs from child to parent
-//
-// the whole chain will be merged into parent, parent will be renamed to child
-// and all the others will deleted
-async function mergeVhdChain(chain, { handler, onLog, remove, merge }) {
-  assert(chain.length >= 2)
+// chain is [ progenitor, child, grand child,  ... , descendant parent, descendant ]
```

Comment:
It's getting too complicated, I'm not sure what they all mean :laughing:

Reply:
challenge accepted 

```
// 1. Create a VhdSynthetic from all children 
// 2. Merge the VhdSynthetic into the ancestor
// 3. Delete all (now) unused VHDs
// 4. Rename the ancestor with the merged data to the latest child
//
//                  VhdSynthetic
//                       |
//              /‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾\
//  [ ancestor, child1, ...,child n-1,  childn ]
//         |     \___________________/   ^
//         |             |               | 
//         |       unused VHDs           |
//         |                             |
//         \__________rename_____________/
```